workers_kv.go

```diff
@@ -0,0 +1,189 @@
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// StorageNamespaceRequest provides parameters for creating and updating storage namespaces
+type StorageNamespaceRequest struct {
+	Title string
+}
+
+// StorageNamespaceResponse is the response received when creating storage namespaces
+type StorageNamespaceResponse struct {
+	Response
+	ID, Title string
+}
+
+// StorageNamespace contains the unique identifier and title of a storage namespace
+type StorageNamespace struct {
+	ID, Title string
+}
+
+// ListStorageNamespacesResponse contains a slice of storage namespaces associated with an
+// account, pagination information, and an embedded response struct
+type ListStorageNamespacesResponse struct {
+	Response
+	Result     []StorageNamespace `json:"result"`
+	ResultInfo `json:"result_info"`
+}
+
+// StorageKey is a key name used to identify a storage value
+type StorageKey struct {
+	Name string `json:"name"`
+}
+
+// ListStorageKeysResponse contains a slice of keys belonging to a storage namespace,
+// paginatiion information, and an embedded response struct
+type ListStorageKeysResponse struct {
+	Response
+	Result     []StorageKey `json:"result"`
+	ResultInfo `json:"result_info"`
+}
+
+// CreateStorageNamespace creates a namespace under the given title.
+// A 400 is returned if the account already owns a namespace with this title.
+// A namespace must be explicitly deleted to be replaced.
+//
+// API reference: https://api.cloudflare.com/#workers-kv-namespace-create-a-namespace
+func (api *API) CreateStorageNamespace(ctx context.Context, req *StorageNamespaceRequest) (StorageNamespaceResponse, error) {
+	uri := fmt.Sprintf("/accounts/%s/storage/kv/namespaces", api.OrganizationID)
+	res, err := api.makeRequestContext(ctx, "POST", uri, req)
+	if err != nil {
+		return StorageNamespaceResponse{}, errors.Wrap(err, errMakeRequestError)
+	}
+
+	result := StorageNamespaceResponse{}
+	if err := json.Unmarshal(res, &result); err != nil {
+		return result, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return result, err
+}
+
+// ListStorageNamespaces lists storage namespaces
+//
+// API reference: https://api.cloudflare.com/#workers-kv-namespace-list-namespaces
+func (api *API) ListStorageNamespaces(ctx context.Context) (ListStorageNamespacesResponse, error) {
+	uri := fmt.Sprintf("/accounts/%s/storage/kv/namespaces", api.OrganizationID)
+	res, err := api.makeRequestContext(ctx, "GET", uri, nil)
+	if err != nil {
+		return ListStorageNamespacesResponse{}, errors.Wrap(err, errMakeRequestError)
+	}
+
+	result := ListStorageNamespacesResponse{}
+	if err := json.Unmarshal(res, &result); err != nil {
+		return result, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return result, err
+}
+
+// DeleteStorageNamespace deletes the namespace corresponding to the given ID
+//
+// API reference: https://api.cloudflare.com/#workers-kv-namespace-remove-a-namespace
+func (api *API) DeleteStorageNamespace(ctx context.Context, namespace string) (Response, error) {
```

Comment:
can this just return `error`?

Reply:
If you would prefer. I'll update this now.

Reply:
Do you want everything that returns (Response, error) to return error instead? Or just certain functions?

Reply:
Discussed IRL, we are going to return a response here.